samples: net: mdns_responder: fix coverity issues

gudnimg opened the pull request:
* Fix #33093 [Coverity CID :215391] Unchecked return value from library in `samples/net/mdns_responder/src/service.c`
* Fix #33094 [Coverity CID :215381] Resource leak in `samples/net/mdns_responder/src/service.c`

jukkar commented on samples/net/mdns_responder/src/service.c:
```diff
@@ -19,11 +19,19 @@ LOG_MODULE_REGISTER(mdns_echo_service, LOG_LEVEL_DBG);
+
+	if (r == -1) {
+		NET_DBG("send() failed (%d)", errno);
+		close(fd);
```
This certainly works, but it would look uniform if the `close()` is done in `service()` so same way as other functions return -1 and then we call close.

GunZi200 replied:
I 100% agree with you. Good idea!

jukkar commented on samples/net/mdns_responder/src/service.c:
```diff
@@ -19,11 +19,19 @@ LOG_MODULE_REGISTER(mdns_echo_service, LOG_LEVEL_DBG);
 /* A default port of 0 causes bind(2) to request an ephemeral port */
```
Commit subject of `samples: net: mdns_responder: Fix coverity issue` is ok, but please change the commit message, so something like this:
```
Check return value if sendto() fails.

Coverity-CID: 215391
Fixes #33093

Signed-off-by: ....
```

GunZi200 replied:
Fixed

jukkar commented on samples/net/mdns_responder/src/service.c:
```diff
@@ -66,7 +66,7 @@ void service(void)
 	/* The simple use case: fixed port */
 	DNS_SD_REGISTER_TCP_SERVICE(zephyr, CONFIG_NET_HOSTNAME,
-				    "_zephyr", "local", DNS_SD_EMPTY_TXT, DEFAULT_PORT);
+					"_zephyr", "local", DNS_SD_EMPTY_TXT, DEFAULT_PORT);
```
This looks like unrelated change and the line is properly indented already, please remove it.

GunZi200 replied:
I had to convert spaces to tabs to fix indentation issues from my changes, it affected the entire file. I will revert this change.

GunZi200 replied:
Fixed

jukkar commented on samples/net/mdns_responder/src/service.c:
```diff
@@ -66,7 +66,7 @@ void service(void)
 #else
```
In commit `samples: net: mdns_responder: Fix coverity resource leak issue`
Same comment about commit message, please change to
```
Avoid resource leak by closing socket when there is an error.

Coverity-CID: 215381
Fixes #33094 

Signed-of-by: ...
```

GunZi200 replied:
Fixed

gudnimg committed:
samples: net: mdns_responder: Fix coverity resource leak issue

Avoid resource leak by closing socket when there is an error.

Coverity-CID: 215381
Fixes #33094

Signed-off-by: Guðni Már Gilbert <gudni.m.g@gmail.com>

gudnimg committed:
samples: net: mdns_responder: Fix coverity issue

Check return value if sendto() fails.

Coverity-CID: 215391
Fixes #33093

Signed-off-by: Guðni Már Gilbert <gudni.m.g@gmail.com>

pfalcon commented on samples/net/mdns_responder/src/service.c:
```diff
@@ -121,7 +121,12 @@ void service(void)
-		welcome(client_fd);
+		r = welcome(client_fd);
+		if (r == -1) {
+			NET_DBG("send() failed (%d)", errno);
```
I'd say this should be NET_ERR().

GunZi200 replied:
@pfalcon Should I also add `NET_ERR()` when `bind()`, `listen()` and `socket()` fail?

GunZi200 replied:
Yeah I used `NET_DBG()` since it's used everywhere else. The main focus is to fix the coverity issue where the return value was not checked. I could add a separate commit to change `NET_DBG()` to `NET_ERR()` on top of the current changes.

@jukkar @tbursztyka what are your thoughts on the matter?

pfalcon replied:
> The main focus is to fix the coverity issue

Right, I remember that, and why I say I don't want to introduce more scope creep into that. LGTM.

jukkar reviewed (approved):
Thanks for the fixes!
